### PR TITLE
clientv3: document context to "Snapshot" API

### PR DESCRIPTION
**Pull request by gyuho**
Address https://github.com/coreos/etcd/issues/8993.

### CODE REVIEW

File: clientv3/maintenance.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ type Maintenance interface {
 	HashKV(ctx context.Context, endpoint string, rev int64) (*HashKVResponse, error)
 
 	// Snapshot provides a reader for a point-in-time snapshot of etcd.
+	// If the context "ctx" is canceled or timed out, reading from returned
+	// "io.ReadCloser" would error, and its error type is "grpc/*status.statusError".
```
Comment (fanminshi) [inline clientv3/maintenance.go]: should we add some sort test to ensure the error message and type is what we believed? I suspect if grpc decides to change underlying error and type, the error message and type could be different.

On the other hand, maybe just let the user knows that ctx cancel/timeout will error out io.ReadCloser is sufficient?
```
// If the context "ctx" is canceled or timed out, reading from returned
// "io.ReadCloser" would error out.
```

Comment (gyuho) [reply clientv3/maintenance.go]: Yeah the error is always `grpc/*status.statusError` type, and we don't have tests around it. Now I think it's better to translate into `context.Canceled` or `context.DeadlineExceeded` as other APIs do. What do you think? /cc @xiang90

Comment (fanminshi) [reply clientv3/maintenance.go]: it seems like this added doc string is too detailed. we don't really have this for other apis. should we just keep the doc string a bit more general?

**Comment by xiang90** [comment]
lgtm.

**Comment by xiang90** [comment]
@gyuho can we get this merged?

**Pull request merged**